### PR TITLE
Frontend package installation with pnpm instructions

### PR DESCRIPTION
**Pull request by joheriks**
Part of vaadin/flow#9719.

### CODE REVIEW

File: articles/guide/installing/frontend.asciidoc [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+You do not need to install pnpm separately.
+Vaadin uses npx, the node package runner to locate (and if necessary download) a compatible pnpm version.
+If you have installed pnpm globally (via `npm install -g pnpm`), the installed version is used by default unless it is determined to be too old.
+
+To install a custom frontend package into your project with pnpm, install Node.js globally and run pnpm using npx.
+For example, to install the `mobx` package into `node_modules`, run the following command in the project directory:
+
+```
+npx pnpm i mobx --shamefully-hoist
```
Comment (haijian-vaadin) [inline articles/guide/installing/frontend.asciidoc:108]: does it also work without the `npx`, i.e. `pnpm i mobx --shamefully-hoist`?

File: articles/guide/installing/frontend.asciidoc [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+= Frontend Tools
+
+This section describes the frontend development tools that Vaadin requires, and some configuration parameters related to the utilization of these.
+
+== Node.js
+
+Vaadin uses the Node.js runtime in development mode to run the Webpack development server, as well as the Node.js package manager (npm) and package runner (npx) to fetch, install and run frontend packages.
+
+pass:[<!-- vale Vaadin.Versions = NO -->]
+As of Vaadin 14.2, Node.js can be installed in three different ways:
```
Comment (jouni) [inline articles/guide/installing/frontend.asciidoc:16]:
```suggestion
Node.js can be installed in three different ways:
```

File: articles/guide/installing/frontend.asciidoc [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+- Automatically into user's home directory (`~/.vaadin/node`).
+- Globally with downloaded installer or package manager (such as Homebrew).
+  Node.js can be downloaded from https://nodejs.org/en/download/[https://nodejs.org/en/download/].
+  Installing Node.js automatically installs the command-line tools `npm` and `npx` as well.
+- Project-local installation (`_project_dir_/node`) using the frontend-maven-plugin[https://github.com/eirslett/frontend-maven-plugin].
+pass:[<!-- vale Vaadin.Versions = YES -->]
```
Comment (jouni) [inline articles/guide/installing/frontend.asciidoc:23]:
```suggestion
```

File: articles/guide/installing/frontend.asciidoc [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+= Frontend Tools
+
+This section describes the frontend development tools that Vaadin requires, and some configuration parameters related to the utilization of these.
+
+== Node.js
+
+Vaadin uses the Node.js runtime in development mode to run the Webpack development server, as well as the Node.js package manager (npm) and package runner (npx) to fetch, install and run frontend packages.
+
+pass:[<!-- vale Vaadin.Versions = NO -->]
```
Comment (jouni) [inline articles/guide/installing/frontend.asciidoc:15]:
```suggestion
```

File: articles/guide/installing/frontend.asciidoc [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+  Node.js can be downloaded from https://nodejs.org/en/download/[https://nodejs.org/en/download/].
+  Installing Node.js automatically installs the command-line tools `npm` and `npx` as well.
+- Project-local installation (`_project_dir_/node`) using the frontend-maven-plugin[https://github.com/eirslett/frontend-maven-plugin].
+pass:[<!-- vale Vaadin.Versions = YES -->]
+
+If Node.js is found globally, Vaadin validates that it is a supported version; if it is too old, it installs a compatible version into `~/.vaadin`.
+We recommend using the latest LTS version.
+A project-local installation will always take precedence over a global or `~/.vaadin` installation.
+
+Older platform versions than 14.2 require manual installation of Node.js.
```
Comment (jouni) [inline articles/guide/installing/frontend.asciidoc:29]:
```suggestion
```

File: articles/guide/installing/frontend.asciidoc [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+You may force Node.js installation to the `~/.vaadin` folder via the `require.home.node` property.
+This property sets the Maven `requireHomeNodeExec` parameter value, so you may configure the Maven goal using `<requireHomeNodeExec>true</requireHomeNodeExec>`.
+To force node installation into home directory in development mode you should use `vaadin.require.home.node` system property or `require.home.node` web init parameter.
+
+
+== PNPM
+
+pass:[<!-- vale Vaadin.Versions = NO -->]
+pnpm reduces the download time across multiple projects by caching the downloaded packages.
+It is the recommended and default package manager for Vaadin 19 projects.
```
Comment (jouni) [inline articles/guide/installing/frontend.asciidoc:96]:
```suggestion
It is the recommended and default package manager for Vaadin projects.
```

File: articles/guide/installing/frontend.asciidoc [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+It is also available for Vaadin 14 as an opt-in; see <<{articles}/flow/advanced/switch-npm-pnpm,Switch Between NPM and NPM>> for more information.
```
Comment (jouni) [inline articles/guide/installing/frontend.asciidoc:97]:
```suggestion
```

File: articles/guide/installing/frontend.asciidoc [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+pass:[<!-- vale Vaadin.Versions = YES -->]
```
Comment (jouni) [inline articles/guide/installing/frontend.asciidoc:98]:
```suggestion
```

File: articles/guide/installing/frontend.asciidoc [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+Please refer to https://docs.travis-ci.com/user/languages/javascript-with-nodejs/#specifying-nodejs-versions[Specifying Node.js versions] in Travis docs how to specify the Node version via `.travis.yml` file.
+
+You may force Node.js installation to the `~/.vaadin` folder via the `require.home.node` property.
+This property sets the Maven `requireHomeNodeExec` parameter value, so you may configure the Maven goal using `<requireHomeNodeExec>true</requireHomeNodeExec>`.
+To force node installation into home directory in development mode you should use `vaadin.require.home.node` system property or `require.home.node` web init parameter.
+
+
+== PNPM
+
+pass:[<!-- vale Vaadin.Versions = NO -->]
```
Comment (jouni) [inline articles/guide/installing/frontend.asciidoc:94]:
```suggestion
```

**Review by jouni — CHANGES_REQUESTED** [review]
Write the docs as if these features have always been part of Vaadin, and assume they will always be part of it.

Documentation for Vaadin 14.x should go to the `v14` branch.

**Comment by haijian-vaadin** [comment]
> Write the docs as if these features have always been part of Vaadin, and assume they will always be part of it.
> 
> Documentation for Vaadin 14.x should go to the `v14` branch.

What if the doc applies to both v14 and latest?

**Comment by jouni** [comment]
> What if the doc applies to both v14 and latest?

Then replicate the same content in both branches, but still avoid explicitly mentioning Vaadin version numbers.

File: articles/guide/installing/frontend.asciidoc (hunk 1 above)
Comment (joheriks) [reply articles/guide/installing/frontend.asciidoc:100]: It does if pnpm is installed globally. Added a note about this and an example.

File: articles/guide/installing/frontend.asciidoc (hunk 2 above)
Comment (joheriks) [reply articles/guide/installing/frontend.asciidoc]: Done.

File: articles/guide/installing/frontend.asciidoc (hunk 3 above)
Comment (joheriks) [reply articles/guide/installing/frontend.asciidoc]: Done.

File: articles/guide/installing/frontend.asciidoc (hunk 4 above)
Comment (joheriks) [reply articles/guide/installing/frontend.asciidoc]: Done.

File: articles/guide/installing/frontend.asciidoc (hunk 5 above)
Comment (joheriks) [reply articles/guide/installing/frontend.asciidoc]: Removed. Also the rest of the paragraph becomes superfluous as it applies to V14.

File: articles/guide/installing/frontend.asciidoc (hunk 6 above)
Comment (joheriks) [reply articles/guide/installing/frontend.asciidoc]: Done.

File: articles/guide/installing/frontend.asciidoc (hunk 7 above)
Comment (joheriks) [reply articles/guide/installing/frontend.asciidoc]: Done.

File: articles/guide/installing/frontend.asciidoc (hunk 8 above)
Comment (joheriks) [reply articles/guide/installing/frontend.asciidoc]: Done.

File: articles/guide/installing/frontend.asciidoc (hunk 9 above)
Comment (joheriks) [reply articles/guide/installing/frontend.asciidoc]: Done.

**Review by jouni — APPROVED** [review]

**Review by platosha — APPROVED** [review]

**Pull request merged**